uitools/toolkitcpp/import/Esri/ArcGISRuntime/Toolkit/MediaPopupElementView.qml

```diff
@@ -43,74 +44,221 @@ ColumnLayout {
+                        tickInterval: Math.abs(listModelData.maxValue) > Math.abs(listModelData.minValue)? max / 3 : Math.abs(min) / 3
+                    }
+
+                    BarSeries {
+                        // BarSeries will take owership of the QList<QBarSet*> when we call append. This also applies to PieSeries and QPieSlice for PieChartPopupMediaItem.
+                        // s.a https://github.com/qt/qtgraphs/blob/ef90c963d10a1af30ed523483e9a589235eccef9/src/graphs2d/barchart/qbarseries.cpp#L670-L692
+                        Component.onCompleted: append(listModelData.barSets);
+                    }
+                }
+            }
+
+            Component {
+                id: pieChartComp
+                GraphsView {
+                    height: thisItem.height
+                    width: thisItem.width
+                    marginTop: -25
+                    marginBottom: -25
+                    marginLeft: -25
+                    marginRight: -25
+
+                    // Setting a default theme if no color information is provided on the PieSeries
+                    theme: GraphsTheme {
+                        backgroundColor: "white"
+                        plotAreaBackgroundColor: "white"
+                        theme: GraphsTheme.Theme.MixSeries
+                    }
+
+                    PieSeries {
+                        // PieSeries will take owership of the QList<QPieSlice*> when we call append. This also applies to BarSeries and QBarSet for BarChartPopupMediaItem.
+                        // s.a https://github.com/qt/qtgraphs/blob/08393457537bb45acf3620986039e516f1ac6f2b/src/graphs2d/piechart/qpieseries.cpp#L635-L672
+                        Component.onCompleted: append(listModelData.pieSlices);
+                    }
+                }
+            }
+
+            Component {
+                id: lineChartComp
+                GraphsView {
+                    height: thisItem.height
+                    width: thisItem.width
+                    marginTop: 10
+                    marginBottom: 10
+                    marginLeft: 10
+                    marginRight: 10
+
+                    // Setting a default theme if no color information is provided on the LineSeries
+                    theme: GraphsTheme {
+                        backgroundColor: "white"
+                        plotAreaBackgroundColor: "white"
+                        theme: GraphsTheme.Theme.MixSeries
+                    }
+
+                    axisX: ValueAxis {
+                        id: xAxis
+                        max: listModelData.linePoints.length - 1
+                        tickInterval: 1
+                        labelsVisible: false
+                    }
+                    axisY: ValueAxis {
+                        id: yAxis
+                        max: listModelData.maxValue
+                        tickInterval: Math.abs(max) > Math.abs(min)? max / 3 : min / 3
```

Comment:
how was the 3 division decided on? does this work well in different cases?

Reply:
Given the initial display being a smaller window(preview if you will) I tested a few values that produced something similar to what I saw on the webmap and 3 was pretty close. When we move to the fullscreen view I am sure it will need changing but given the available screen real-estate, 3 looks good to me. I should have added some pictures to the PR title comment. I will add some for reference.